### ISSUE TITLE
Add scaffolding and remove gitignore for settings.php

### ISSUE THREAD

**Issue by mikemadison13** [acquia/drupal-recommended-project#73]
For whatever reason, we are gitignoring the docroot/sites/default/settings.php file. I believe this shouldn't be the case (and I always have to force add it).

Also, for those using this scaffolding w/o BLT, there isn't currently anything that will automatically generate a settings.php file. We should probably add a composer script to copy default.settings.php to settings.php.

**Comment by balsama** [comment]
> For whatever reason, we are gitignoring the docroot/sites/default/settings.php file.
I don't believe this is the case anymore. See 5a73dd629f7e0defcecf0fd7b010b46e72017501. Do we have an error in our .gitignore, or is your project from before #63 was merged?

> there isn't currently anything that will automatically generate a settings.php file
I'm not a huge fan of adding (many) scripts to composer.json. Would adding an ACLI command be acceptable?

**Comment by mikemadison13** [comment]
@balsama 

@pavithra-raman182 and I just saw as of today that settings.php is still gitignored. that's why in the PR i changed the rules, i think https://github.com/acquia/drupal-recommended-project/blob/master/.gitignore#L8 is still hitting the normal settings.php file (because i have to alter this line every time or force commit it)

i'm trying to anticipate users who aren't taking advantage of BLT. so if ACLI has some sort of initial project scaffolding / generation thing, then sure i'm open to that. my thinking on the composer scaffold was that we are already copying the default files out of core and into sites/default, it seems trivial (and sensible) to _also_ create the real versions of those files, since the default ones are absolutely worthless unless you know to copy them (or are using an automation tool like BLT to put them in place).

**Comment by mikemadison13** [comment]
TLDR we have people using this project and ending up without their settings files in place (so things don't work in CI/CD or on the cloud). i think the project should create the stock files by default. period (although I do agree with you in general that it's a slippery slope and we shouldn't go too far down it)

**Comment by balsama** [comment]
>  I just saw as of today that settings.php is still gitignored

Ah. This is because the latest tag ([1.8.2](https://github.com/acquia/drupal-recommended-project/blob/v1.8.2/.gitignore)) was cut before #63 was merged. We just need a new release to get the fix.

**Comment by balsama** [comment]
We'll work on getting some other PRs merged in the morning and release out ASAP after that.

**Comment by mikemadison13** [comment]
just closing the loop here, and confirming that the gitignore in the current release does in fact allow the default settings.php file to be committed.
